crates/ffi/src/erc6492_client.rs

```diff
@@ -0,0 +1,46 @@
+use crate::ffi::Erc6492Error;
+use alloy::{
+    network::Ethereum,
+    primitives::{Address, Bytes, B256},
+    providers::ReqwestProvider,
+};
+
+pub struct Erc6492Client {
+    provider: ReqwestProvider<Ethereum>,
+}
+
+impl Erc6492Client {
+    pub fn new(rpc_url: String) -> Self {
+        let url = rpc_url.parse().expect("Invalid RPC URL");
+        let provider = ReqwestProvider::<Ethereum>::new_http(url);
+        Self { provider }
+    }
+
+    pub async fn verify_signature(
```

Comment:
I assume for Kotlin it would be copy and paste this method

Reply:
Yeah